app/src/main/java/com/github/se/polyfit/ui/screen/AdditionalMealInfoScreen.kt

```diff
@@ -0,0 +1,79 @@
+package com.github.se.polyfit.ui.screen
+
+import android.util.Log
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.BottomAppBar
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.polyfit.model.meal.MealOccasion
+import com.github.se.polyfit.ui.components.scaffold.SimpleTopBar
+import com.github.se.polyfit.ui.components.selector.DateSelector
+import com.github.se.polyfit.ui.components.selector.MealOccasionSelector
+import com.github.se.polyfit.ui.components.selector.MealTagSelector
+import com.github.se.polyfit.ui.theme.PrimaryPurple
+import com.github.se.polyfit.viewmodel.meal.MealViewModel
+
+@Composable
+fun AdditionalMealInfoScreen(
+    mealViewModel: MealViewModel,
+    navigateBack: () -> Unit,
+    navigateForward: () -> Unit
+) {
+  val meal = mealViewModel.meal.observeAsState()
+  val occasion = meal.value?.occasion ?: MealOccasion.OTHER
+  val mealTags = meal.value?.tags ?: mutableListOf()
+
+  // Date will default to today, and tags are optional
+  val isComplete = occasion != MealOccasion.OTHER
+
+  Scaffold(
+      topBar = { SimpleTopBar("Additional Meal Info", navigateBack) },
+      bottomBar = { BottomBar(isComplete, navigateForward) }) {
+        Column(
+            modifier = Modifier.padding(it).fillMaxWidth().testTag("AdditionalMealInfoContainer")) {
+              DateSelector(onConfirm = mealViewModel::setMealCreatedAt)
+              MealOccasionSelector(occasion, mealViewModel::setMealOccasion)
+              MealTagSelector(
```

Comment:
## Issue: UI Scaling with Multiple Tags

**Problem Description:**
There is an  issue with the UI for the `Special Tags` when there is a large number of tags. Currently, adding many tags causes the bottom tabs to become inaccessible, this could be due an overflow issue with the tag container.

**Screenshot:**
![Screenshot of UI Issue](https://github.com/swent-group10/polyfit/assets/100274930/7dcb5acd-6dbd-47c9-9172-4759b534c819)

**Proposed Solutions:**
**Scrollable Tag Container:** Implement scrolling for the tags area to keep the bottom tabs accessible, no matter how many tags are displayed/added.

Reply:
I'm wondering if we should just limit the amount of tags: i.e. 5, 10. At a certain point its a bit redundant and someone could fill up our DB with a bunch of trash

Reply:
If that works with you, I'll leave it as a ticket in the back log for the future. Gonna try to shift focus towards the day summary UI

Reply:
actually ill just add the limit in now